agents/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/launcher/AbstractAgentLauncher.java

```diff
@@ -66,13 +68,33 @@ public AbstractAgentLauncher(long agentMaxStartTimeMs,
         this.agentLaunchingChecker = agentLaunchingChecker;
     }
 
+    /**
+     * Launches agent script in machine and waits until agent starts.
+     * The machine should be started.
+     * Retrieves agent's logs if its start fails.
+     *
+     * @param machine
+     *         the machine instance in which agent will be started. The machine should be started.
+     * @param agent
+     *         the agent to start
+     * @throws ServerException
+     *         if script execution failed
+     */
     @Override
     public void launch(Instance machine, Agent agent) throws ServerException {
         if (isNullOrEmpty(agent.getScript())) {
             return;
         }
+        ListLineConsumer agentLogger = new ListLineConsumer();
+        LineConsumer lineConsumer = new AbstractLineConsumer() {
```

Comment:
Possible, but in that case we shouldn't close `CompositeLineConsumer`, only `agentLogger` (which is part of composite consumer).